rskj-core/src/main/java/co/rsk/core/bc/ReadWrittenKeysTracker.java

```diff
@@ -19,20 +19,27 @@
 package co.rsk.core.bc;
 
 import org.ethereum.db.ByteArrayWrapper;
-
+import java.util.HashMap;
 import java.util.HashSet;
+import java.util.Map;
 import java.util.Set;
 
 //TODO(JULI):
 // * Next step should be to check whether a key is written in the cache but also deleted in the same transaction. This key shouldn't be considered as a written key.
 
 public class ReadWrittenKeysTracker implements IReadWrittenKeysTracker {
+    private final Map<ByteArrayWrapper, Long> threadByReadKey;
+    private final Map<ByteArrayWrapper, Long> threadByWrittenKey;
     private Set<ByteArrayWrapper> temporalReadKeys;
     private Set<ByteArrayWrapper> temporalWrittenKeys;
```

Comment:
You can use the keyset of each map as the set of read/written keys. There is no need to keep separate Sets.